elixir/apps/domain/lib/domain/auth/adapter/openid_connect/directory_sync.ex

```diff
@@ -382,6 +385,43 @@ defmodule Domain.Auth.Adapter.OpenIDConnect.DirectorySync do
     end)
   end
 
+  defp maybe_email_sync_error(provider) do
+    with {:ok, provider} <-
+           Domain.Auth.fetch_active_provider_by_id(provider.id, preload: :account) do
+      send_sync_error_email(provider)
+    else
+      _ ->
+        Logger.warning("Failed to fetch provider while attempting to send sync error email")
+    end
```

Comment:
Will `Repo.preload` also update the `provider`?

Reply:
Nevermind, I played around with it and see that it doesn't.  The reason I was fetching the provider again was because the provider may not contain the last sync errors.

I did see that there is a `Repo.reload`, but it doesn't seem to allow preloads from what I've found, but I may have just overlooked something.

I'm realizing that we don't always record the error for a directory sync error it seems, which is ultimately what led me to need to refresh the provider.  Maybe we can sync on this and discuss the best way to proceed.